cmd/earthly/subcmd/build_cmd.go

```diff
@@ -500,21 +500,28 @@ func (a *Build) ActionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs []stri
 	}
 
 	cacheImports := make([]string, 0)
+	var cacheImportImageName string
 	if a.cli.Flags().RemoteCache != "" {
-		cacheImports = append(cacheImports, a.cli.Flags().RemoteCache)
+		cacheImportImageName, _, err = a.parseImageNameAndAttrs(a.cli.Flags().RemoteCache)
+		cacheImports = append(cacheImports, cacheImportImageName)
 	}
 	if len(a.cacheFrom.Value()) > 0 {
 		cacheImports = append(cacheImports, a.cacheFrom.Value()...)
 	}
 	var cacheExport string
 	var maxCacheExport string
+	var cacheExportAttrs map[string]string
 	if a.cli.Flags().RemoteCache != "" && a.cli.Flags().Push {
 		if a.cli.Flags().MaxRemoteCache {
-			maxCacheExport = a.cli.Flags().RemoteCache
+			maxCacheExport, cacheExportAttrs, err = a.parseImageNameAndAttrs(a.cli.Flags().RemoteCache)
```

Comment:
since the max cache setting ends up in a different variable name, I would create a similar one for attributes
```suggestion
			maxCacheExport, maxCacheExportAttrs, err = a.parseImageNameAndAttrs(a.cli.Flags().RemoteCache)
```

(alternatively I would remove the `MaxCacheExport` field entirely, and just set the `max` attribute to true, but please don't do this in this PR, since it might be a larger task)

Reply:
I'm just unsure if there's a chance the code was designed to support both a min and max export mode; that's why I'm wary of making an even bigger change in this PR.

Reply:
I agree with you on this point.

P.S. Honestly, I would recommend considering deprecating the `--max-remote-cache` option and just adding a flag like: `--export-attributes`, it would be easier from my point of view.

Reply:
About your commit suggestion, it will lead to creating additional fields 
```
	maxCacheExportAttrs map[string]string
```
in `builder` and `solver` structs, is that okay for you?

Reply:
yes that is what I'm asking for, thanks.

Reply:
Okay, I'll do it now :)

Reply:
@alexcb done